Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -199,8 +201,9 @@ editLocation rideId (_, merchantId) req = do
   whenJust req.origin $ \pickup -> do
     when (ride.status /= SRide.NEW) do
       throwError (InvalidRequest $ "Customer is not allowed to change pickup as the ride is not NEW for rideId: " <> ride.id.getId)
-    lastMapping <- QLM.findLastMapping ride.id.getId 0 >>= fromMaybeM (InternalError $ "Latest mapping not found for rideId: " <> ride.id.getId)
-    initialLocationForRide <- QL.findById lastMapping.locationId >>= fromMaybeM (InternalError $ "Location not found for locationId:" <> lastMapping.locationId.getId)
+    x <- QLM.findAllByEntityIdAndOrder ride.id.getId 0
+    oldestMapping <- (listToMaybe $ sortBy (comparing (Down . (.order))) x) & fromMaybeM (InternalError $ "Latest mapping not found for rideId: " <> ride.id.getId)
```

Comment:
Search is done for `order = 0` and then sorting is also done based on order only. What fix is this PR doing and how is this code helping?